### ISSUE TITLE
Error 500 - Internal Server Error: System.DivideByZeroException: Attempted to divide by zero.

### ISSUE THREAD

**Issue by 0xB10C** [btcpayserver/btcpayserver#3747]
**Describe the bug**
When going to the main site of my BTCPayServer (should show a crowdfunding page), it returns a Error 500 - Internal Server Error.

**To Reproduce the bug**
Steps to reproduce the reported bug:
1. Go to https://pay.b10c.me
2. See error 500

**Expected behavior**

I'd expect to see the crowdfunding page.

**Your BTCPay Environment (please complete the following information):**
- BTCPay Server Version: 1.5.1
- Deployment Method:  Manual / via https://github.com/fort-nix/nix-bitcoin
- Browser:  Tested in Firefox and Chrome

**Logs (if applicable)**
```
info: Configuration:  Data Directory: <REDACTED>
info: Configuration:  Configuration File: <REDACTED>
info: BTCPayServer.Plugins.PluginManager: Loading plugins from <REDACTED>
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer - 1.5.1.0
info: BTCPayServer.Plugins.PluginManager: Adding and executing plugin BTCPayServer.Plugins.Shopify - 1.5.1.0
info: Configuration:  Supported chains: BTC
info: Configuration:  BTC: Explorer url is <REDACTED>
info: Configuration:  BTC: Cookie file is <REDACTED>
info: Configuration:  Network: Mainnet
info: Configuration:  Root Path: /
info: PayServer:      Starting listening NBXplorer (BTC)
info: PayServer:      Start watching invoices
info: PayServer:      Starting payment request expiration watcher
info: Configuration:  Now listening on: <REDACTED>
info: PayServer:      0 pending payment requests being checked since last run
info: PayServer:      BTC: Checking if any pending invoice got paid while offline...
info: PayServer:      BTC: 0 payments happened while offline
info: PayServer:      Connected to WebSocket of NBXplorer (BTC)
info: PayServer:      User '<REDACTED>' logged in.
warn: PayServer:      Unhandled exception in AppHubStreamer
System.DivideByZeroException: Attempted to divide by zero.
   at System.Decimal.DecCalc.VarDecDiv(DecCalc& d1, DecCalc& d2)
   at BTCPayServer.Services.Apps.AppService.GetInfo(AppData appData)
   at BTCPayServer.Services.Apps.AppService.GetAppInfo(String appId) in /build/source/BTCPayServer/Services/Apps/AppService.cs:line 58
   at BTCPayServer.Services.Apps.AppHubStreamer.InfoUpdated(String appId) in /build/source/BTCPayServer/Services/Apps/AppHubStreamer.cs:line 58
   at BTCPayServer.Services.Apps.AppHubStreamer.ProcessEvent(Object evt, CancellationToken cancellationToken) in /build/source/BTCPayServer/Services/Apps/AppHubStreamer.cs:line 52
   at BTCPayServer.HostedServices.EventHostedServiceBase.ProcessEvents(CancellationToken cancellationToken) in /build/source/BTCPayServer/HostedServices/EventHostedServiceBase.cs:line 44
fail: Microsoft.AspNetCore.Server.Kestrel: Connection id "<REDACTED>", Request id "<REDACTED>": An unhandled exception was thrown by the application.
System.DivideByZeroException: Attempted to divide by zero.
   at System.Decimal.DecCalc.VarDecDiv(DecCalc& d1, DecCalc& d2)
   at BTCPayServer.Services.Apps.AppService.GetInfo(AppData appData)
   at BTCPayServer.Services.Apps.AppService.GetAppInfo(String appId) in /build/source/BTCPayServer/Services/Apps/AppService.cs:line 58
   at BTCPayServer.Controllers.UIAppsPublicController.GetAppInfo(String appId) in /build/source/BTCPayServer/Controllers/UIAppsPublicController.cs:line 411
   at BTCPayServer.Controllers.UIAppsPublicController.ViewCrowdfund(String appId, String statusMessage) in /build/source/BTCPayServer/Controllers/UIAppsPublicController.cs:line 278
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|26_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ExceptionContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at BTCPayServer.Hosting.BTCPayMiddleware.Invoke(HttpContext httpContext) in /build/source/BTCPayServer/Hosting/BTCpayMiddleware.cs:line 101
   at BTCPayServer.Hosting.GreenfieldMiddleware.Invoke(HttpContext httpContext) in /build/source/BTCPayServer/Hosting/GreenfieldMiddleware.cs:line 50
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at BTCPayServer.Hosting.HeadersOverrideMiddleware.Invoke(HttpContext httpContext) in /build/source/BTCPayServer/Hosting/HeadersOverrideMiddleware.cs:line 30
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
fail: Microsoft.AspNetCore.Server.Kestrel: Connection id "<REDACTED>", Request id "<REDACTED>": An unhandled exception was thrown by the application.
```

**Additional context**

Did work fine before setting a newly created crowdfunding page (as my old stores + app got lost after upgrade to 1.5.1) as starting page.

**Comment by NicolasDorier** [comment]
it is impossible your app got lost after an upgrade, we don't do anything to delete data.

About the exception, looking to it. It seems it is because the `TargetAmount` of crowdfund is 0.

**Comment by NicolasDorier** [comment]
I fixed this one in 1.5.3!
